napari/_qt/widgets/qt_viewer_status_bar.py

```diff
@@ -129,3 +105,115 @@ def _toggle_activity_dock(self, visible: Optional[bool] = None):
         else:
             par._activity_dialog.hide()
             self._activity_item._activityBtn.setArrowType(Qt.ArrowType.UpArrow)
+
+
+class StatusBarWidget(QWidget):
+    def __init__(
+        self,
+        status_label: QLabel,
+        layer_label: QLabel,
+        source_label: QLabel,
+        plugin_label: QLabel,
+        coordinates_label: QLabel,
+        help_label: QLabel,
+        parent: QWidget = None,
+    ):
+        super().__init__(parent=parent)
+        self._status_label = status_label
+        self._layer_label = layer_label
+        self._source_label = source_label
+        self._plugin_label = plugin_label
+        self._coordinates_label = coordinates_label
+        self._help_label = help_label
+
+        self._status_label.setParent(self)
+        self._layer_label.setParent(self)
+        self._source_label.setParent(self)
+        self._plugin_label.setParent(self)
+        self._coordinates_label.setParent(self)
+        self._help_label.setParent(self)
+
+    def resizeEvent(self, event: QResizeEvent) -> None:
+        super().resizeEvent(event)
+        self.do_layout()
+
+    def event(self, event: QEvent) -> bool:
+        if event.type() == QEvent.Type.LayoutRequest:
+            self.do_layout()
+        return super().event(event)
+
+    @staticmethod
+    def _calc_width(fm: QFontMetrics, label: QLabel) -> int:
+        # magical nuber +2 is from superqt code
+        # magical number +6 is from experiments
```

Comment:
What do these magic numbers do? Increase the base width a bit?

Reply:
Ah, makes sense :)

Reply:
except there is no +6 in the code, only +8! 😂 Could you maybe elaborate in the comments based on this discussion?

Reply:
I need to bump the value to 12 after merging main. I hate this, and I do not know how to do it in a proper way...